### PR TITLE
reduce number of top-level files

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- blocks #177 https://github.com/krassowski/jupyterlab-lsp/pull/177#discussion_r370933583

## Code changes

Moves a lot of things around. There is a hard reference to `LANGSERVERS.md` in the source: we should probably just inline that or something... or point to `readthedocs` when we do that.

## User-facing changes

The repo is a tad cleaner on first blush:
- fits on a full screen, meh
  - Screenshot from 2020-01-25 15-19-16

## Backwards-incompatible changes

_shouldn't_ create any issues on PRs...

## Chores

- [x] linted
- [ ] tested
- [ ] documented
- [ ] changelog entry

### CODE REVIEW

**Comment by bollwyvl** [comment]
To _really_ go after it, we need all the meta junk from python:
- `setup.{cfg,py}`
- `MANIFEST.in`
and js:
- `package.json`
- `lerna.json`
- `yarn.lock`
- `.prettierignore`
.. in separate dirs altogether. I'm happy to do that (once I get at least _this_ working), but that will start to impact the usability, i fear.

**Comment by bollwyvl** [comment]
Another thought: if the DX is indeed getting onerous, we can fix it Once And For All and introduce a top-level automation tool, and indeed, that might even be worthwhile, especially as we increase the number of built artifacts (#177, #163). If we want to go that way, this PR is a _fine_ place to do it: i find it onerous to incrementally change automation if we can get consensus vision of where we want to go.

`make` would be great, but in 2020, it's still hard to reliably get a working windows toolchain, same goes for node-based stuff, so i'd favor [doit](https://pydoit.org/). Others i've tried include `pants` and `scons`: again, they are rather heavy opinions.

Positing `doit` (but likely very similar for other tools): this would replace `scripts` with a single, top-level `dodo.py`, and after a `pip install -e .` (and getting node somehow), you'd type `doit <thing>`. Running without any args would basically run the entire azure pipeline locally, which is actually _fantastic_ for reviewing PRs and CI. If you do a touch of extra work, it can do the make-like thing: e.g. if you change the ts, but don't touch the python, it won't even run the python unit tests, but _would_ run the acceptance tests (after first rebuilding your `lab` so you don't get the `thinger blocked by dialog` warning).

Anyhow, food for thought...

**Comment by bollwyvl** [comment]
Well, I guess I'll take it passing the first time. No doubt i missed some stuff in the markdown links, but that will be tested aggressively on #177, so less inclined to worry about it at this point...

File: .prettierrc [hunk 1]
```diff
@@ -1,3 +0,0 @@
-{
-  "singleQuote": true
```
Comment (krassowski) [inline .prettierrc]: `.prettierrc` got lost in the transition. It was there for interoperability of the codebase with the core (https://github.com/jupyterlab/jupyterlab/blob/master/.prettierrc). Probably could go to `packages`.

File: package.json [hunk 2]
```diff
@@ -28,6 +28,9 @@
       "pre-commit": "precise-commits"
     }
   },
+  "prettier": {
```
Comment (bollwyvl) [inline package.json]: This is where `.prettierrc` went!

https://github.com/krassowski/jupyterlab-lsp/pull/180#discussion_r370960292

File: packages/jupyterlab-lsp/src/adapters/jupyterlab/components/statusbar.tsx [hunk 3]
```diff
@@ -211,7 +211,7 @@ class LSPPopup extends VDomRenderer<LSPStatus.Model> {
           Documentation:{' '}
           <a
             href={
-              'https://github.com/krassowski/jupyterlab-lsp/blob/master/LANGUAGESERVERS.md'
+              'https://github.com/krassowski/jupyterlab-lsp/blob/master/docs/LANGUAGESERVERS.md'
```
Comment (krassowski) [inline packages/jupyterlab-lsp/src/adapters/jupyterlab/components/statusbar.tsx]: One lazy option would be to point to a version-specific commit and have a link to master in the document itself (something like "there might  be an updated version of this document >click here<".)

**Pull request merged**